immupdate.ts

```diff
@@ -158,8 +158,13 @@ class _Updater {
     let modified = structurallyModified
 
     if (value === DELETE) {
-      
-      if (leafHostIsOption) {
+
+      if (Array.isArray(leafHost) && typeof field === 'number') {
+        if (field >= leafHost.length) return target
```

Comment:
modifying this line to `if (field >= leafHost.length || (field == leafHost.length && value !== DELETE)) return target` should give the user the ability to append an element to the array.